src/Voron/Exceptions/VoronUnrecoverableErrorException.cs

```diff
@@ -10,7 +10,7 @@
 
 namespace Voron.Exceptions
 {
-    public class VoronUnrecoverableErrorException : Exception
+    public class VoronUnrecoverableErrorException : ApplicationException
```

Comment:
Why do we need that?

Reply:
Lucene is wrapping the exception with a SystemException except of IOException, SystemException and ApplicationException

https://github.com/ravendb/lucenenet/blob/3.0.3-netcore/src/Lucene.Net/Index/IndexWriter.cs#L4636

Reply:
Can you please add a comment here explaining what is the reason? Otherwise it's easy to forget about it

Reply:
Nevermind, we'll have it in github